internal/service/moduleconfig/reader/moduleconfig_reader.go

```diff
@@ -0,0 +1,200 @@
+}
+
+func (s *Service) GetDefaultCRData(defaultCRPath string) ([]byte, error) {
+	if defaultCRPath == "" {
+		return nil, nil
+	}
+	defaultCRData, err := s.fileSystem.ReadFile(defaultCRPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read default CR file: %w", err)
+	}
+
+	return defaultCRData, nil
+}
+
+func (s *Service) CleanupTempFiles() []error {
+	return s.tmpFileSystem.RemoveTempFiles()
+}
+
+func GetManifestPath(manifestPath string, tempFileSystem TempFileSystem) (string, error) {
+	path := manifestPath
+
+	if parsedURL, err := ParseURL(manifestPath); err == nil {
+		path, err = tempFileSystem.DownloadTempFile("", defaultManifestFilePattern, parsedURL)
+		if err != nil {
+			return "", fmt.Errorf("failed to download Manifest file: %w", err)
+		}
+		return path, nil
+	}
+
+	if !filepath.IsAbs(manifestPath) {
+		// Get the current working directory
+		homeDir, err := os.Getwd()
+		if err != nil {
+			return "", fmt.Errorf("failed to get the current directory: %w", err)
+		}
+		// Get the relative path from the current directory
+		path = filepath.Join(homeDir, path)
+		path, err = filepath.Abs(path)
+		if err != nil {
+			return "", fmt.Errorf("failed to obtain absolute path to manifest file: %w", err)
+		}
+		return path, nil
+	}
+
+	return path, nil
+}
+
+func ParseURL(urlString string) (*url.URL, error) {
+	urlParsed, err := url.Parse(urlString)
+	if err == nil && urlParsed.Scheme != "" && urlParsed.Host != "" {
+		return urlParsed, nil
+	}
+	return nil, fmt.Errorf("%w: parsing url failed for %s", commonerrors.ErrInvalidArg, urlString)
+}
+
+func ValidateModuleConfig(moduleConfig *contentprovider.ModuleConfig) error {
+	if err := validation.ValidateModuleName(moduleConfig.Name); err != nil {
+		return fmt.Errorf("failed to validate module name: %w", err)
+	}
+
+	if err := validation.ValidateModuleVersion(moduleConfig.Version); err != nil {
+		return fmt.Errorf("failed to validate module version: %w", err)
+	}
+
+	if err := validation.ValidateModuleChannel(moduleConfig.Channel); err != nil {
+		return fmt.Errorf("failed to validate module channel: %w", err)
+	}
+
+	if err := validation.ValidateModuleNamespace(moduleConfig.Namespace); err != nil {
+		return fmt.Errorf("failed to validate module namespace: %w", err)
+	}
+
+	if moduleConfig.ManifestPath == "" {
+		return fmt.Errorf("%w: manifest path must not be empty", commonerrors.ErrInvalidOption)
+	}
+
+	return nil
+}
+
+func ParseModuleConfig(configFilePath string, fileSystem FileSystem) (*contentprovider.ModuleConfig, error) {
+	moduleConfigData, err := fileSystem.ReadFile(configFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read module config file: %w", err)
+	}
+
+	moduleConfig := &contentprovider.ModuleConfig{}
+	if err := yaml.Unmarshal(moduleConfigData, moduleConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse module config file: %w", err)
+	}
+
+	if moduleConfig.Namespace == "" {
+		moduleConfig.Namespace = "kcp-system"
+	}
```

Comment:
good point. I believe it is no longer needed since we will drop the one cluster setup and the remote ModuleTemplate feature is already deprecated. But we can remove this in a separate PR. Here is the issue for it: https://github.com/kyma-project/modulectl/issues/50